src/lib/proxy-server.ts

```diff
@@ -0,0 +1,189 @@
+		return null;
+	}
+}
+
+/**
+ * Attempts to start the proxy server on port 80
+ * This requires admin/root privileges
+ */
+export async function startProxyServer(): Promise< boolean > {
+	if ( isProxyRunning ) return true;
+
+	try {
+		// First, try starting on a privileged port directly (might work if app has permissions)
+		const result = await startDomainProxy();
+		if ( result ) {
+			return true;
+		}
+
+		// If direct start fails and we're not skipping elevation, attempt with sudo/admin privileges
+		return await startProxyWithElevation();
+	} catch ( error ) {
+		console.error( 'Failed to start proxy server:', error );
+		return false;
+	}
+}
+
+/**
+ * Attempts to start the proxy server directly (without privilege elevation)
+ * @param {number} port - The port to listen on (defaults to 80)
+ */
+async function startDomainProxy( port: number = 80 ): Promise< boolean > {
+	try {
+		// Create proxy with additional options to preserve host header
+		const proxy = httpProxy.createProxyServer();
+
+		proxy.on( 'error', ( err, req, res ) => {
+			console.error( 'Proxy error:', err );
+			if ( res && 'writeHead' in res ) {
+				res.writeHead( 500 );
+				res.end( 'Proxy error: ' + err.message );
+			}
+		} );
+
+		proxyServer = http.createServer( async ( req, res ) => {
+			const host = req.headers.host?.split( ':' )[ 0 ]; // Remove port if present
+
+			// Log incoming request information for debugging
+			console.log( `Received request with host header: ${ host }` );
+
+			// Look up the port directly from user data
+			if ( ! host ) {
+				console.log( 'No host header found' );
+				res.writeHead( 404 );
+				res.end( 'No host header found' );
+				return;
+			}
+
+			const port = await getPortForDomain( host );
+			if ( ! port ) {
+				console.log( `Domain not found: ${ host }` );
+				res.writeHead( 404 );
+				res.end( `Domain not found: ${ host }` );
+				return;
+			}
+
+			console.log( `Proxying request for ${ host } to port ${ port }` );
+
+			// Forward the request with the original host preserved
+			proxy.web( req, res, {
+				target: `http://localhost:${ port }`,
+				xfwd: true, // Pass along x-forwarded headers
+			} );
+		} );
+
+		await new Promise< void >( ( resolve, reject ) => {
+			proxyServer!
+				.listen( port, () => {
+					console.log( `Proxy server started on port ${ port }` );
+					isProxyRunning = true;
+					resolve();
+				} )
+				.on( 'error', ( err ) => {
+					console.error( `Error starting proxy server on port ${ port }:`, err );
+					reject( err );
+				} );
+		} );
+
+		return true;
+	} catch ( error ) {
+		console.error( `Failed to start proxy server directly:`, error );
+		return false;
+	}
+}
+
+/**
+ * Attempts to start the proxy server with elevated privileges
+ */
+async function startProxyWithElevation(): Promise< boolean > {
+	const currentPlatform = platform();
+
+	try {
+		console.log( 'Attempting to start proxy server with elevated privileges' );
+
+		if ( currentPlatform === 'win32' ) {
```

Comment:
Apparently in Windows, we need to run the proxy with admin rights. The AI proposed using `netsh` to do so. I need someone to help me test Windows.